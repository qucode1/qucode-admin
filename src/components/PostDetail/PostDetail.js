import React from 'react'
import Post from '../Post/Post'
import NotFound from '../NotFound/NotFound'
import { Route } from 'react-router-dom'

const PostDetail = function (props) {
  const matchingPost = props.posts.posts.find((post) => post.slug === props.match.params.slug)
  return (
    <div>
      {matchingPost
        ? <Post {...matchingPost} details />
        : <Route component={() => <NotFound message='Post not found' />} />}
    </div>
  )
}

export default PostDetail
