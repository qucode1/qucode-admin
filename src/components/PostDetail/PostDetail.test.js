import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { MemoryRouter } from 'react-router-dom'

import PostDetail from './PostDetail'

const posts = {
  'posts': [{
    'slug': 'post-slug'
  }]
}

const match = {
  'params': {
    'slug': 'post-slug'
  }
}

const mismatch = {
  'params': {
    'slug': 'different-slug'
  }
}

test('Should render PostDetails when visiting /posts/:slug', () => {
  const component = mount(
    <MemoryRouter initialEntries={['/posts/testing-with-jest-and-enzyme']} initialIndex={0} >
      <PostDetail posts={posts} match={match}/>
    </MemoryRouter>
  )

  expect(component.find(PostDetail).length).toBe(1)
})

it('PostDetail should render as expected', () => {
  const wrapper = shallow(<PostDetail posts={posts} match={match}/>)
  const tree = toJson(wrapper)
  expect(tree).toMatchSnapshot()
})

it('PostDetail should render as expected', () => {
  const wrapper = shallow(<PostDetail posts={posts} match={mismatch}/>)
  const tree = toJson(wrapper)
  expect(tree).toMatchSnapshot()
})
