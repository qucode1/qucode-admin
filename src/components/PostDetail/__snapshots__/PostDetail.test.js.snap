// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetail should render as expected 1`] = `
<div>
  <Post
    details={true}
    slug="post-slug"
  />
</div>
`;

exports[`PostDetail should render as expected 2`] = `
<div>
  <Route
    component={[Function]}
  />
</div>
`;
